### ISSUE TITLE
Do not bundle install unused dependencies from `licensee`

### ISSUE THREAD

**Issue by jonabc** [github/licensed#66]
With https://github.com/github/licensed/pull/65, we've removed any immediate dependency on `gem octokit`.

However, running `bundle install` still installs octokit, rugged and their dependencies even though those gems are entirely unused by `licensed`.

**Comment by jonabc** [comment]
It looks like those gems can be explicitly set to not install with bundler by putting them in a group and then running `bundle install --without <group>`.

When I try this locally it seems to work for octokit and it's dependencies but not for rugged.  Looking through Gemfile.lock, I only see `rugged` being used by `licensee`.  I took a look at `licensee` and noticed that `octokit` is autoloaded while `rugged` is not.  I opened https://github.com/benbalter/licensee/pull/314 to see if that would allow rugged to not be installed.

Not installing rugged would be 💖 , as it's native extensions take more time to install then the rest of the dependencies combined.  Installing fewer gems also means the packaged executables should be smaller, which is nice when the linux archive package is a chunky ~30MB.
